x-pack/plugins/spaces/server/lib/audit_logger.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export class SpacesAuditLogger {
+  private readonly enabled: boolean;
+  private readonly auditLogger: any;
+
+  constructor(config: any, auditLogger: any) {
+    this.enabled = config.get('xpack.security.audit.enabled');
+    this.auditLogger = auditLogger;
+  }
+  public spacesAuthorizationFailure(username: string, action: string, spaceIds?: string[]) {
+    if (!this.enabled) {
+      return;
+    }
+
+    this.auditLogger.log(
+      'spaces_authorization_failure',
```

Comment:
Make sure to add this to the audit logging docs in docs/security/audit-logging.asciidoc